client/components/select-dropdown/test/index.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import { assert } from 'chai';
+import { shallow, mount } from 'enzyme';
+import sinon from 'sinon';
```

Comment:
Yeah, I've noticed it during my exploration of your test base; it is essentially a _mocha-aware_ wrapper around `sinon.sandbox.create`.
Are you suggesting it's better to use it even in a such simple scenario?

Also, I just noticed that for the assertions on spy/stub you never use `sinon.assert`.
I usually prefer `sinon.assert` cause it's more descriptive than chai, qunit (whatever...) when an assertion fails.
Do you see any problem?

Finally, I would ask @gziolo a question about `use-sinon`.
I noticed that the pattern:
```js
let sandbox;
useSandbox( newSandbox => {
    sandbox = newSandbox;
});
```
is diffused through the whole testbase.

I was wondering why `useSandbox` does not just return the newly created sandbox; and noticed that this functionality was recently added (https://github.com/Automattic/wp-calypso/commit/80727d5ca7b9198632605452f35c84ab3097a8a6), and then removed (https://github.com/Automattic/wp-calypso/commit/f37aa64d22da48f4722b9612fd9814e3b758ab85#diff-c988aa06ee204545fc3c01efb3ffe8c6).

I think the idea was pretty cool; however the initial implementation had a flaw (and probably the reason you removed it) in that it returned an empty object when executing `useSandbox`. That was due to the fact that `sinon.sandbox.create( config );` was called too late (when mocha triggers the before hook).

I tried a different implementation, and tests still pass.
If in the next days I prepare a pull request, do you mind to review it?